docs/DeveloperGuide.md

```diff
@@ -1,38 +1,228 @@
+
+#### Entering incomplete information
+Test case: `delete expense` or `delete expense id/12`\
+Expected: Terminal shows error message.
+
+### Edit
+#### Edit past expense
+1. Prerequisites:
+   - Ensure that a valid expense id is entered
+   - Ensure that a valid date is entered
+   - Date is entered in the format: yyyymmdd
+   - Ensure all fields are entered
+   
+2. Test case:
+   `edit expense id/1 in/"amount/date/category"`\
+   Enter type of the field to edit in the above format. \
+   Expected: Terminal shows message to enter the information for the chosen field to edit. Successful message is shown on entering the information.
+
+#### Edit future expense
+1. Prerequisites:
+   - Ensure that a valid expense id is entered
+   - Ensure that a valid date is entered
+   - Date is entered in the format: yyyymmdd
+   - Ensure all fields are entered
+
+2. Test case:
+   `edit future expense id/1 in/"amount/date/category"`\
+   Enter type of the field to edit in the above format. \
+Expected: Terminal shows message to enter the information for the chosen field to edit. Successful message is shown on entering the information.
+
+### List
+#### List past expense
+Test case:
+   `list expenses`\
+   Expected: Terminal shows message to enter the number associated with the sorting type of expenses (by date/name/amount). List of expenses is displayed based on the entered sorting type. If expense list is empty, terminal displays required message.
+
+#### List future expense
+Test case:
+`list future expenses`\
+Expected: Terminal shows list of upcoming payments with the total amount due and current balance. A warning message is displayed if the balance is insufficient.
+
+### Balance
+#### Set balance
+1. Prerequisites:
+   - Ensure all fields are entered
+   
+2. Test case:
+`set balance $/2000`\
+Expected: Terminal shows successful message.
+
+#### Check balance
+Test case:
+`check balance`\
+Expected: Terminal shows remaining balance.
+
+#### Entering incomplete information
+Test case: `set balance`\
+Expected: Terminal shows error message.
+
+### Upcoming Expenses
+#### Check upcoming expenses
+Test case:
+`check upcoming expenses`\
+Expected: Terminal shows message to choose period of upcoming expenses. Upcoming payments for the entered time period are displayed with the total amount due and current balance. A warning message is displayed if the balance is insufficient.
+
+#### Pay upcoming expenses
+1. Prerequisites:
+   - Ensure that a valid future expense id is entered
+   - Ensure all fields are entered
+2. Test case:
+`pay 1`\
+Expected: Terminal shows successful message. The paid future expense is added to list of past expenses.
+
+
+### Shutdown
+After the testing is done, type the command `bye` to exit the program\
+Terminal will display a goodbye message and the application would close in the command-line interface.
```

Comment:
Maybe can show examples of what the output warning messages are for each feature